### ISSUE TITLE
Generate better API documentation

### ISSUE THREAD

**Issue by brendandahl** [mozilla/pdf.js#6526]
The api docs at https://mozilla.github.io/pdf.js/api/draft/ are really bad. We have a lot of good information in api.js and in other files but it doesn't seem to be parsed correctly. Things we should look into:

- fixing the docs in the js file so they are more compatible with jsdoc
- add some more examples within the docs
- document the viewer api
- update the theme so it includes our gh-pages header/footer

If anyone is willing to look into this, I'm happy to mentor and/or quickly review pull requests.

**Comment by prometheansacrifice** [comment]
@brendandahl I'd like to take this up. Can you point me to the files that have incompatible comments?

**Comment by brendandahl** [comment]
Yeah, there are a number of comments that aren't showing up in the generated documentation.

**Comment by prometheansacrifice** [comment]
@brendandahl PR [6599] (https://github.com/mozilla/pdf.js/pull/6599) fixes missing comments.

Go ahead with adding examples?

**Comment by mennucc** [comment]
also, the examples in `https://mozilla.github.io/pdf.js/examples/` do not work:

- they refer to `PDFJS` whereas it is now called `pdfjsLib`
- it refers to `compatibility.js` which does not exist
- `renderTask.then(function () ...` fails

**Comment by EndlessGoodness** [comment]
I would like to take up this issue. How may I start on this? @brendandahl
